DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -36,8 +36,14 @@ public IActionResult Index(int customisationId)
                 return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
             }
 
-            var progressId = courseContentService.GetProgressId(User.GetCandidateId(), customisationId);
-            courseContentService.UpdateProgress(progressId);
+            var progressId = courseContentService.GetOrCreateProgressId(User.GetCandidateId(), customisationId, User.GetCentreId().Value);
+
+            if (progressId == null)
+            {

```

Comment:
Worth logging an error here I think 🙂

Reply:
The service should have already logged the error, should I log it again?

Reply:
Yes I think it's worth it so there's a nice trace of errors in the logs. Something like "Redirecting to 500 as no progress id returned for user ..." 🙂 We should probably add a similar log when we're redirecting to 404 actually